src/NuGet.Core/NuGet.Build.Tasks/RestoreSettingsUtils.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.Common;
+using NuGet.Configuration;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace NuGet.Build.Tasks
+{
+    public class RestoreSettingsUtils
```

Comment:
Add a high level description for this class explaining that this handles falling back to nuget.config settings or reading project properties